herd_behaviour.nlogo

```diff
@@ -1,72 +1,53 @@
-;; avoidance range radius
-;; following range radius
-;; percent_informed - proportion of informed individuals
-;; weight - value between 0 and 2 to weight there own direction preference
-;; number_herd_members - number of herd members
-;; aim-direction - d-dir in paper
+
 globals [
-  number_informed
-  number_uninformed
-  g-dir ;; preffered direction same for all informed turtles same
-  w ;; weight for all informed turtles same
+  ;; avoidance_range
+  ;; following_range
+  ;; percent_informed - proportion of informed individuals
+  ;; number_herd_members - number of herd members
+  ;; weigth - weight with which prefered direction is included
 ]
 
 turtles-own [
-  ;; want to maintain minimum distancs alpha
-  speed ;; individual speed
-  omega
+  speed ;; individual speed (modelled on agent level to add different speeds later)
   v-dir ;; direction vector
-  informed ;; boolean
-  ;; ci (position vector)
-  neighbours-avoid ;; neighours in alpha avoidance range
-  neighbours-follow ;; neighbours in follow range
+  temp-v-dir ;; temporary direction vector that is calculated and applied in the end whe all turtles have calcultated their directions
+  g-dir ;; preffered direction there can be multiple prefered directions
+  w ;; weight with which prefered direction is included
+  ;; c (implicit position vector)
 ]
 
 to setup
   clear-all
   reset-ticks
-  set w weight
-  set g-dir aim-direction
-  set number_informed (number_herd_members * ( 0.01 * (percent_informed)))
-  set number_uninformed (number_herd_members - number_informed)
-  create-turtles number_informed [
+  create-turtles number_herd_members [
     set xcor -100
     set ycor -100
-    set informed true
+    set color blue
+    set w 0
   ]
-  create-turtles number_uninformed [
-    set xcor -100
-    set ycor -100
-    set informed false
+  let number_informed_members (number_herd_members * ( 0.01 * (percent_informed)))
+  ask n-of number_informed_members turtles [
+    set w  weight
+    set color red
   ]
 end
 
 to go
   ask turtles[
-    move
+    ifelse count other turtles in-radius avoidance_range > 0 [
+      ;; avoid them
+      calculate-dir
+    ][
+      if count other turtles in-radius following_range > 0[
+       ;;follow them
+        calculate-dir-with-w-g
+      ]
+    ]
+    forward 1
   ]
   tick
 end
 
-to move
-  find-neighbours-avoid
-  find-neighbours-follow
-  if any? neighbours-avoid
-    [ calculate-dir ] ;; formula 1
-  if any? neighbours-follow
-    [ calculate-dir-with-v ] ;; formula 2
```

Comment:
I think there is a difference between formula 2 and 3 not just with the weight, let's talk about it on Wednesday pls